HRTool/src/main/java/app/controllers/ReportRestController.java

```diff
@@ -61,6 +58,18 @@
 	@Autowired
     private ServletContext servletContext;
 	
+	/**
+	 * Returns only the tag clouds that contain the given employee
+	 * @param extracted
+	 * @param toExtract
+	 * @param employee
+	 * @return
+	 */
+	private void extractClouds(Set<TagCloud> extracted, List<TagCloud> toExtract, Employee employee){
+		toExtract.stream().filter(tc -> tc.getEmployees().contains(employee)).forEach(tc -> extracted.add(tc));

```

Comment:
Please use full variable names if applicable.
Are you using reverse logic? Instead of getting all tagClouds for an employee, you're running through a list of tagClouds and checking if an employee has it?